OpenOversight/app/models.py

```diff
@@ -518,6 +518,12 @@ def password(self):
     def password(self, password):
         self.password_hash = generate_password_hash(password, method="pbkdf2:sha256")
 
+    def by_email(email):
+        return User.query.filter(func.lower(User.email) == func.lower(email))
+
+    def by_username(username):
+        return User.query.filter(func.lower(User.username) == func.lower(username))
+
```

Comment:
Potential refactor just to reuse code, up to you on which you prefer:
```suggestion
    def _case_insensitive_equality(field, value):
        return User.query.filter(func.lower(field) == func.lower(value))

    def by_email(email):
        return _case_insensitive_equality(User.email, email)

    def by_username(username):
        return _case_insensitive_equality(User.username, username)

```